### PR TITLE
Fix e2e test case2 status flaky issue

### PR DESCRIPTION
**Pull request by yiraeChristineKim**

### CODE REVIEW

**Comment by yiraeChristineKim** [comment]
stable

**Review by mprahl — CHANGES_REQUESTED** [review]

File: controllers/common/common_status_update.go [hunk 1]
```diff
@@ -50,8 +50,20 @@ func RootStatusUpdate(ctx context.Context, c client.Client, rootPolicy *policies
 		return decisions, nil
 	}
 
-	log.Info("Updating the root policy status", "RootPolicyName", rootPolicy.Name, "Namespace", rootPolicy.Namespace)
+	log.Info("Sorting the root policy status", "RootPolicyName", rootPolicy.Name, "Namespace", rootPolicy.Namespace)
+	sort.SliceStable(placements, func(i, j int) bool {
```
[the remark this answers is not in the document]
Comment (yiraeChristineKim) [reply controllers/common/common_status_update.go:54]: OK! good

**Review by mprahl — DISMISSED** [review]

File: controllers/common/common_status_update.go [hunk 2]
```diff
@@ -44,14 +44,26 @@ func RootStatusUpdate(ctx context.Context, c client.Client, rootPolicy *policies
...
+		pi := placements[i].PlacementBinding + " " + placements[i].Placement + " " +
+			placements[i].PlacementRule + " " + placements[i].PolicySet
+		pj := placements[j].PlacementBinding + " " + placements[j].Placement + " " +
+			placements[j].PlacementRule + " " + placements[j].PolicySet
+
+		return pi < pj
+	})
+	sort.SliceStable(cpcs, func(i, j int) bool {
+		return cpcs[i].ClusterName < cpcs[j].ClusterName
+	})
```
Comment (JustinKuli) [inline controllers/common/common_status_update.go:58]: This list is already sorted in `CalculatePerClusterStatus`. If that needs to be updated to be a stable sort, I think it should be updated there.

Maybe sorting the `placements` should also be done in its function... but it doesn't make a big difference.

Comment (yiraeChristineKim) [reply controllers/common/common_status_update.go:58]: maybe I can remove this. `placements` which function you mean?

Comment (yiraeChristineKim) [reply controllers/common/common_status_update.go:58]: `GetClusterDecisions` ?

**Review by JustinKuli — APPROVED** [review]

File: controllers/common/common_status_update.go [hunk 3]
```diff
@@ -236,6 +235,16 @@ func GetClusterDecisions(
 		}
 	}
 
+	log.Info("Sorting placements", "RootPolicyName", rootPolicy.Name, "Namespace", rootPolicy.Namespace)
```
Comment (mprahl) [inline controllers/common/common_status_update.go:238]: This shouldn't be an info log message.
```suggestion
	log.V(2).Info("Sorting placements", "RootPolicyName", rootPolicy.Name, "Namespace", rootPolicy.Namespace)
```

**Review by mprahl — APPROVED** [review]

**Pull request merged**